test/unit/test_jinja.py

```diff
@@ -1,11 +1,378 @@
+        '''foo: "'{{ true }}'"''',
+        returns("'True'"),
+        returns("'True'"),
+    ),  # jinja true -> python True -> str(True) -> "True" -> quoted
+    (
+        '''foo: "'{{ true | as_text }}'"''',
+        returns("'True'"),
+        returns("'True'"),
+    ),
+    (
+        '''foo: "'{{ true | as_bool }}'"''',
+        returns("'True'"),
+        returns("'True'"),
+    ),
+    (
+        '''foo: "'{{ true | as_number }}'"''',
+        returns("'True'"),
+        returns("'True'"),
+    ),
+    (
+        '''foo: "'{{ true | as_native }}'"''',
+        returns("'True'"),
+        returns("'True'"),
+    ),
+    # unquoted True
+    (
+        '''foo: "{{ True }}"''',
+        returns('True'),
+        returns('True'),
+    ),
+    (
+        '''foo: "{{ True | as_text }}"''',
+        returns("True"),
+        returns("True"),
+    ),  # True -> string 'True' -> text -> str('True') -> 'True'
+    (
+        '''foo: "{{ True | as_bool }}"''',
+        returns("True"),
+        returns(True),
+    ),
+    (
+        '''foo: "{{ True | as_number }}"''',
+        returns("True"),
+        raises(JinjaRenderingException),
+    ),
+    (
+        '''foo: "{{ True | as_native }}"''',
+        returns("True"),
+        returns(True),
+    ),
+    # quoted 'True' within rendering
+    (
+        '''foo: "{{ 'True' | as_text }}"''',
+        returns("True"),
+        returns("True"),
+    ),
+    # 'True' -> string 'True' -> text -> str('True') -> 'True'
+    (
+        '''foo: "{{ 'True' | as_bool }}"''',
+        returns('True'),
+        returns(True),
+    ),
+    # quoted 'True' outside rendering
+    (
+        '''foo: "'{{ True }}'"''',
+        returns("'True'"),
+        returns("'True'"),
+    ),
+    (
+        '''foo: "'{{ True | as_bool }}'"''',
+        returns("'True'"),
+        returns("'True'"),
+    ),
+    # yaml turns 'yes' into a boolean true
+    (
+        '''foo: yes''',
+        returns('True'),
+        returns('True'),
+    ),
+    (
+        '''foo: "yes"''',
+        returns('yes'),
+        returns('yes'),
+    ),
+    # concatenation
+    (
+        '''foo: "{{ (a_int + 100) | as_native }}"''',
+        returns('200'),
+        returns(200),
+    ),
+    (
+        '''foo: "{{ (a_str ~ 100) | as_native }}"''',
+        returns('100100'),
+        returns(100100),
+    ),
+    (
+        '''foo: "{{( a_int ~ 100) | as_native }}"''',
+        returns('100100'),
+        returns(100100),
+    ),
+    # multiple nodes -> always str
+    (
+        '''foo: "{{ a_str | as_native }}{{ a_str | as_native }}"''',
+        returns('100100'),
+        returns('100100'),
+    ),
+    (
+        '''foo: "{{ a_int | as_native }}{{ a_int | as_native }}"''',
+        returns('100100'),
+        returns('100100'),
+    ),
+    (
+        '''foo: "'{{ a_int | as_native }}{{ a_int | as_native }}'"''',
+        returns("'100100'"),
+        returns("'100100'"),
+    ),
+    (
+        '''foo:''',
+        returns('None'),
```

Comment:
This one is surprising to me - why do we return a string `'None'` instead of a literal `None` here? This test differs from my experiences with exercising this PR on the CLI. When I test either of the following values, I get a literal `None` back instead of a stringy `'None'`
```
foo:
foo: null
```

Reply:
dbt doesn't ever pass non-str values to `get_rendered`, so this aspect of the test doesn't reflect what dbt will actually do during rendering (see `dbt.config.renderer.BaseRenderer.render_value`).

I wanted to make sure we did have the underlying behavior documented in unit tests, at least. There are integration tests that exercise the behavior you're talking about.

Reply:
ahhh, that makes total sense too - thanks